wordpress-mobile/WordPress-Android
# Restoring the usage of commitAllowingStateLoss instead of commit in Bottom Nav

Fixes #13987 

<details>
  <summary>History left here for context; please look the update section below for final decisions</summary>
  
Was looking into the above mentioned issue and drafting this PR to further discuss and double check the approach we can use to fix (pinging you since you worked on the code involved and may have some context 🙇 )

AFAIU we were using `commitAllowingStateLoss` when setting the fragment page on the bottom navigation until some point like here below:

```
if (fragment != null) {
            fragmentManager
                    .beginTransaction()
                    .replace(R.id.fragment_container, fragment, getTagForPosition(position))
                    // This is used because the main activity sometimes crashes because it's trying to switch fragments
                    // after `onSaveInstanceState` was already called. This is the related issue
                    // https://github.com/wordpress-mobile/WordPress-Android/pull/10853
                    .commitAllowingStateLoss()
        }
```

in the specific case of this issue (attempt to login from reader bottom sheet when being in a self-hosted account) that was preventing the crash that can happen considering the timing of [this related part of code ](https://github.com/wordpress-mobile/WordPress-Android/blob/ef56b66e13cb50f2edc5217536f7755002179257/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java#L583).

You can forcely reproduce the issue setting to 1500 ms the current 500 ms delay there.

With this increased delay and using the `commitAllowingStateLoss´I was not able to spot odd conditions around that specific scenario or smoke testing; wondering if we could restore that behavior. wdyt? any insight I'm missing?

Another possibility could be to remove the `mBottomNav.postDelayed` and setting the bottom navigation to My Site when coming back from Me section. This could be maybe cleaner but tbh I'm not sure it's trivial to collect/discriminate all possible cases when you should set the My Site page or not 🤔.

Thanks for your thoughts/insights 🙇‍♂️
</details>


## UPDATE

We are keeping the commit and relying on View Model live data lifecycle awareness, separating the start of the login flow from the setting of main tab as main page. Re this last point, we found (at least) one flow that do not set the main page to my site. Talking about when the user actually decide to login. In this case the VM is restarted so we need to preventively set the main page to be My Site in the shared prefs.

### To test
- Starting from a self-hosted user, go to the reader subfilter
- Tap the "Log in to WordPress.com" 
- Check you are landed into Me section and no crash
- Come back and check you are landed into the My Site page
- Do the same steps as above but now do a login in the Me Site
- After the login check you are landed in the My Site page and not into the reader page that was the last selected page by the user

PR submission checklist:
